staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -343,7 +343,7 @@ func ConnectWithRedirects(originalMethod string, originalLocation *url.URL, head
 
 redirectLoop:
 	for redirects := 0; ; redirects++ {
-		if redirects > maxRedirects {
+		if redirects >= maxRedirects {
```

Comment:
first request isn't a redirect, so I think `>` is correct... `maxRedirects=1` would let you go through the loop twice

Reply:
It should fail on the 10th redirect, not allow 10 redirects, at least to be consistent with http.Client. I changed this because the test caught this inconsistency.

Reply:
I would expect maxRedirects=10 to allow 10 and fail on 11. I don't think it really matters in practice either way as long as we don't set maxRedirects to 1

Reply:
I'd like to keep this consistent with the default http redirect checker, so I reverted the logic but changed maxRedirects to 9.  (Also fixed the spdy test with the new max)